cpp/daal/src/externals/service_math_ref.h

```diff
@@ -58,48 +58,88 @@ struct RefMath<double, cpu>
 
     static double sPowx(double in, double in1) { return pow(in, in1); }
 
+    static double xsPowx(double in, double in1) { return pow(in, in1); }
+
     static double sCeil(double in) { return ceil(in); }
 
+    static double xsCeil(double in) { return ceil(in); }
+
     // Not implemented
     static double sErfInv(double in) { return std::numeric_limits<double>::quiet_NaN(); }
 
+    // Not implemented
+    static double xsErfInv(double in) { return std::numeric_limits<double>::quiet_NaN(); }
```

Comment:
I haven't looked at what exact compilation arguments go into the `icx` invocations, but isn't the default for it to have the equivalent of GCC's `-ffast-math` which implicitly assumes that NaNs and Infs will not be encountered?

If so, would this NaN be checked somewhere? Could it throw an exception instead?

And how about taking these functions from the `cephes` library from Netlib which is public domain?